src/kiwixapp.cpp

```diff
@@ -34,6 +35,10 @@ KiwixApp::KiwixApp(int& argc, char *argv[])
     }
     qInfo() << "Compiled with Qt Version " << QT_VERSION_STR;
     qInfo() << "Runtime Qt Version " << qVersion();
+
+    QFuture<void> future = QtConcurrent::run([=](){
+        m_library.syncNewBooksInLibrary(kiwix::getDataDirectory());
+    });
```

Comment:
We have to be careful with multithreading. The library is not thread safe, so you must protect it from multiple call with a mutex.